experimental/dds/tree2/src/feature-libraries/editable-tree-2/editableTreeTypes.ts

```diff
@@ -125,6 +125,11 @@ export interface TreeNode extends Tree<TreeSchema> {
 		listener: EditableTreeEvents[K],
 	): () => void;
 
+	/**
+	 * Gets a field of this node.
+	 */
+	getField(key: FieldKey): TreeField;
```

Comment:
In the current implementation, it was a specific design choice not to provide any APIs while allow getting fields which are under arbitrary (not in the schema) keys on non-map nodes.

For example, if allow this, and try and fix our field lifetime rules so field object identity is stable, struct, leaf and field nodes will need to cache the results of this so they can be reused on future calls in a map, then probably do some weak ref stuff to avoid that leaking. We would also have to decide what the field schema for such out of schema fields is (probably FieldSchema.empty): currently there is no way to get a field that doesn't have an explicit field schema, but adding this method changes that.

Anyway, how valuable is this method to you?